Add Specialized Turbo integration docs (#44119)
home-assistant/home-assistant.github.io: source/_integrations/specialized_turbo.markdown

---
title: Specialized Turbo
description: Read telemetry from Specialized Turbo e-bikes over Bluetooth Low Energy in Home Assistant.
ha_category:
  - Sensor
ha_release: "2026.4"
ha_iot_class: Local Push
ha_config_flow: true
ha_codeowners:
  - '@JamieMagee'
ha_domain: specialized_turbo
ha_integration_type: device
ha_quality_scale: bronze
ha_platforms:
  - sensor
ha_bluetooth: true
related:
  - docs: /docs/configuration/troubleshooting/#debug-logs-and-diagnostics
    title: Debug logs and diagnostics
  - docs: /integrations/bluetooth/
    title: Bluetooth
---

The **Specialized Turbo** {% term integration %} connects Home Assistant to [Specialized](https://www.specialized.com/) Turbo e-bikes over Bluetooth Low Energy. It reads real-time telemetry straight from the bike — battery level, speed, power output, cadence, motor temperature, odometer, and more — giving you the same data the Specialized Mission Control app sees.

If you ride a Specialized Turbo Vado, Levo, Creo, Como, or Tero, you can track battery health over time, log rides through Home Assistant's history, or build automations that fire when your bike finishes charging.

## Supported devices

> This integration works with Specialized Turbo e-bikes that have the Gen 2 BLE module (the "TURBOHMI2017" protocol), which includes most 2017 and newer models with a <abbr title="Turbo Connect Unit">TCU</abbr> display:

Reviewer: ```suggestion
This integration works with Specialized Turbo e-bikes that have the Gen 2 BLE module (the TURBOHMI2017 protocol), which includes most 2017 and newer models with a <abbr title="Turbo Connect Unit">TCU</abbr> display:
```

- Turbo Vado (all SL and full-power variants)
- Turbo Levo (all SL and full-power variants)
- Turbo Creo (all SL and full-power variants)
- Turbo Como (SL and full-power)
- Turbo Tero

> If your bike broadcasts a "TURBOHMI" Bluetooth advertisement when powered on, it should work.

Reviewer: ```suggestion
If your bike broadcasts a TURBOHMI Bluetooth advertisement when powered on, it should work.
```

- Third-party e-bikes

## Prerequisites

Before setting up this integration, make sure:

1. Your bike is powered on and awake (pedal or press the power button — the TCU screen should be lit).
2. Home Assistant has access to a [Bluetooth adapter](/integrations/bluetooth/). If the bike is out of range, you can use an [ESPHome Bluetooth proxy](https://esphome.io/projects/?type=bluetooth) placed in your garage or shed.
3. No other app (like Specialized Mission Control) is actively connected to the bike over Bluetooth. Only one BLE client can connect at a time.

{% include integrations/config_flow.md %}

The bike is usually discovered automatically. If it doesn't appear, make sure it's powered on and within Bluetooth range, then try adding it manually through {% my integrations title="**Settings** > **Devices & services**" %}.

{% configuration_basic %}
Pairing PIN:
  description: "The PIN displayed on the bike's TCU screen during Bluetooth pairing. Leave empty if your bike does not require a PIN."
{% endconfiguration_basic %}

## Supported functionality

### Entities

The **Specialized Turbo** integration provides the following sensor entities.

#### Sensors

- **Battery**

>   - **Description**: Current battery charge level, in percent.

Reviewer: ```suggestion
  - **Description**: Current battery charge level, as a percentage.
```

- **Sleep mode**: When the bike enters sleep mode, Bluetooth advertisements stop and the connection drops. Data resumes when you wake the bike up.
- **Secondary battery**: Range-extender battery data is parsed internally but not exposed as sensor entities.

## Troubleshooting

### Bike not discovered

If your bike doesn't appear during setup:

1. Make sure the bike is powered on and awake — pedal briefly or press the power button so the TCU screen lights up.
2. Check that your Bluetooth adapter is working in {% my integrations title="**Settings** > **Devices & services**" %}.
3. If you're using an ESPHome Bluetooth proxy, confirm that `active: true` is set in the proxy's configuration.
4. Move the bike closer to the Bluetooth adapter. BLE range is typically 5–10 meters, less through walls.

### Sensors show "Unavailable"

This usually means the bike is out of range or asleep:

1. Wake the bike by pedaling or pressing the power button.
2. Check whether the Specialized Mission Control app (or another BLE client) has an active connection — only one client can connect at a time.

> 3. Try restarting the integration from {% my integrations title="**Settings** > **Devices & services**" %}.

Reviewer: ```suggestion
### I can´t see sensor data [ what can´t I see / access / do? ]

#### Symptom: sensors are unavailable

[what happens in the UI? what is the message I get?]

#### Description

This usually means the bike is out of range or asleep. [cause]

#### Resolution

To resolve this issue, try the following steps:

1. Wake the bike by pedaling or pressing the power button.
2. Check whether the Specialized Mission Control app (or another BLE client) has an active connection — only one client can connect at a time.
3. Try restarting the integration from {% my integrations title="**Settings** > **Devices & services**" %}.
```

### Pairing PIN not accepted

The PIN is displayed on the bike's TCU screen when it enters pairing mode. If you don't see a PIN prompt on the bike, it may already be paired with another device. Reset the BLE pairing on the bike and try again.

> Some Bluetooth backends don't support programmatic PIN entry. If that happens, pair the bike through your operating system's Bluetooth settings first, then set up the integration without entering a PIN.

Reviewer: ```suggestion
### Bike does not enter pairing mode

#### Symptom: Pairing PIN is not requested

When trying to enter pairing mode, I don´t see a PIN prompt on the bike.

#### Description

The PIN is displayed on the bike’s TCU screen when it enters pairing mode. If you don’t see a PIN prompt on the bike, the reason can be one of the following:

- The bike was already paired with another device.
- The Bluetooth backends don´t support programmatic PIN entry.

#### Resolution

To resolve the issue, do one of the following depending on the reason above respectively:

- Reset the BLE pairing on the bike and try again
- Pair the bike through your operating system’s Bluetooth settings first, then set up the integration without entering a PIN.
```

If you ride a Specialized Turbo Vado, Levo, Creo, Como, or Tero, you can track battery health over time, log rides through Home Assistant's history, or build automations that fire when your bike finishes charging.

## Supported devices

This integration works with Specialized Turbo e-bikes that have the Gen 2 BLE module (the TURBOHMI2017 protocol), which includes most 2017 and newer models with a <abbr title="Turbo Connect Unit">TCU</abbr> display:

- Turbo Vado (all SL and full-power variants)
- Turbo Levo (all SL and full-power variants)
- Turbo Creo (all SL and full-power variants)
- Turbo Como (SL and full-power)
- Turbo Tero

If your bike broadcasts a TURBOHMI Bluetooth advertisement when powered on, it should work.

## Unsupported devices

- Older Specialized bikes without a TCU display
- Non-Turbo Specialized bikes (road, mountain, gravel without a motor)
- Third-party e-bikes

## Prerequisites

Before setting up this integration, make sure:

> 1. Your bike is powered on and awake (pedal or press the power button — the TCU screen should be lit).

Reviewer: ```suggestion
1. Your bike is powered on and awake. If it isn´t, pedal or press the power button. The TCU screen should be lit.
```

- **Sleep mode**: When the bike enters sleep mode, Bluetooth advertisements stop and the connection drops. Data resumes when you wake the bike up.
- **Secondary battery**: Range-extender battery data is parsed internally but not exposed as sensor entities.

## Troubleshooting

### Bike not discovered

#### Symptom: The bike doesn't appear during setup

When you try to add the integration, your bike is not listed as a discovered device.

#### Description

The bike may be asleep, out of Bluetooth range, or the Bluetooth adapter in Home Assistant may not be picking up its advertisement.

#### Resolution

To resolve this issue, try the following steps:

1. Make sure the bike is powered on and awake. Pedal briefly or press the power button so the TCU screen lights up.
2. Check that your Bluetooth adapter is working in {% my integrations title="**Settings** > **Devices & services**" %}.
3. If you're using an ESPHome Bluetooth proxy, confirm that `active: true` is set in the proxy's configuration.
4. Move the bike closer to the Bluetooth adapter. BLE range is typically 5–10 meters, less through walls.

> ### Sensors show "Unavailable"

Reviewer: To avoid repetition in titles, I would suggest here something like:
```suggestion
### Can´t see the values of sensor entities
```

#### Symptom: All sensors show "Unavailable"

> After the integration is set up, one or more sensor entities show **Unavailable** instead of a value.

Reviewer: It seems there is an inconsistency here. The text mentions "one or more sensor entities" but the title says "All sensors".

Author: Thanks, I changed the body to "all" as well.

  - **Description**: Configured assistance for Trail mode
  - **Category**: Diagnostic
  - **Remarks**: Disabled by default

- **Turbo assist**
  - **Description**: Configured assistance for Turbo mode
  - **Category**: Diagnostic
  - **Remarks**: Disabled by default

- **Range (long)**
  - **Description**: Estimated range for the long-range calculation
  - **Remarks**: Disabled by default

- **Range (short)**
  - **Description**: Estimated range for the short-range calculation
  - **Remarks**: Disabled by default

- **Altitude**
  - **Description**: Current altitude
  - **Remarks**: Disabled by default

- **Altitude gain**
  - **Description**: Total altitude gain
  - **Remarks**: Disabled by default

- **Gradient**
  - **Description**: Current gradient in percent
  - **Remarks**: Disabled by default

- **System temperature**
  - **Description**: Temperature reported by the bike system
  - **Category**: Diagnostic
  - **Remarks**: Disabled by default

- **Consumption**
  - **Description**: Energy consumption in watt-hours per kilometer
  - **Remarks**: Disabled by default

- **Calories**
  - **Description**: Total ride energy in kilocalories
  - **Remarks**: Disabled by default

> ## Examples

Reviewer: ```suggestion
## Automation examples
```

### Notify when the battery is fully charged

```yaml
- alias: "Bike battery full"
  triggers:
    - trigger: numeric_state

>       entity_id: sensor.specialized_turbo_battery
>       above: 99
>   actions:
>     - action: notify.mobile_app
>       data:
>         message: "Your bike is fully charged."

Reviewer: ```suggestion
  actions:
    - action: notify.send_message
      target:
        entity_id: notify.my_device
      data:
        message: "Your bike is fully charged."
```

```

### Warn when battery health decreases

{% raw %}

```yaml
- alias: "Bike battery health warning"
  triggers:
    - trigger: numeric_state
      entity_id: sensor.specialized_turbo_battery_health
      below: 80
  actions:
    - action: notify.mobile_app
      data:
        message: >
          Bike battery health is at
          {{ states(
            'sensor.specialized_turbo_battery_health'
          ) }}%.

>           Consider scheduling a service.

Reviewer: ```suggestion
    - action: notify.send_message
      target:
        entity_id: notify.my_device
      data:
        message: >
          Bike battery health is at
          {{ states(
            'sensor.specialized_turbo_battery_health'
          ) }}%.
          Consider scheduling a service.
```
